lib/origen/commands/compile.rb

```diff
@@ -43,7 +43,16 @@
   opts.on('-r', '--reference DIR', String, 'Override the default reference directory') { |o| options[:reference] = o }
   opts.on('-z', '--zip', 'Gzip all output files (diff checking will be skipped)') { |o| options[:zip] = o }
   app_options.each do |app_option|
-    opts.on(*app_option) {}
+    if app_option.last.is_a?(Proc)
+      ao_proc = app_option.pop
+      if ao_proc.arity == 1
+        opts.on(*app_option) { instance_exec(options, &ao_proc) }
```

Comment:
Do we even need the instance_exec, given that we are passing in the options arg?

This would be a more common approach, though I don't know what instance_exec does, maybe its the same:

~~~ruby
opts.on(*app_option) { ao_proc.call(options) } 
~~~

Reply:
Yes, that works too.  I switched it over.